atomic_reactor/plugins/input_osv3.py

```diff
@@ -186,3 +187,68 @@ def run(self):
     @classmethod
     def is_autousable(cls):
         return 'BUILD' in os.environ and 'SOURCE_URI' in os.environ and 'OUTPUT_IMAGE' in os.environ
+
+
+class OSv3SourceContainerInputPlugin(InputPlugin):
+    """Input plugin for building source container images"""
+    key = "osv3_source_container"
+
+    def __init__(self, **kwargs):
+        """
+        constructor
+        """
+        # call parent constructor
+        super(OSv3SourceContainerInputPlugin, self).__init__(**kwargs)
+        self.target_registry = None
+        self.reactor_env = None
+        self.plugins_json = None
+
+    def run(self):
+        build_json = get_build_json()
+        image = os.environ['OUTPUT_IMAGE']
+        self.target_registry = os.environ.get('OUTPUT_REGISTRY', None)
+
+        arrangement_version = None
+        try:
+            user_params = os.environ['USER_PARAMS']
+            user_data = json.loads(user_params)  # TODO validate user params against schema
+            arrangement_version = user_data.get('arrangement_version', None)
+            self.plugins_json = get_plugins_with_user_data(user_params, user_data)
+            # if we get the USER_PARAMS, we'd better get the REACTOR_CONFIG too
+            reactor_config_map = os.environ['REACTOR_CONFIG']
+            self.reactor_env = read_yaml(reactor_config_map, 'schemas/config.json')
+        except KeyError:
```

Comment:
remove this, since we won't have that ATOMIC_REACTOR_PLUGINS variable defined for sources for sure

Reply:
removed